### DISCUSSION TITLE
Unable to deploy plain YAMLs using Kustomization example

### DISCUSSION THREAD

**Discussion by cnukwas** [fluxcd/flux2#3475] [Q&A] [answered]
Hi,

I'm newbie to Flux and able to install/sync Helm releases but seeing an issue when I tried to sync plain YAMLs by following [plan yamls](https://fluxcd.io/flux/faq/#can-i-use-repositories-with-plain-yamls) example in the documentation. I see that GitRepository and Kustomization resources are created and seeing following error in kustomize controller logs and and when I describe the resource as well. Using Flux2 v0.38.2.

I am not sure why the error is referring to a different CRD from kustomize.config.k8s.io/v1beta1. I really appreciate any help to resolve this issue.

```
kubectl -n flux-system describe kustomizations.kustomize.toolkit.fluxcd.io testdeploy
Name:         testdeploy
Namespace:    flux-system
API Version:  kustomize.toolkit.fluxcd.io/v1beta2
Kind:         Kustomization
Metadata:
Spec:
  Force:     false
  Interval:  10m0s
  Path:      ./releases/testdeploy
  Prune:     true
  Source Ref:
    Kind:  GitRepository
    Name:  testdeploy
Status:
  Conditions:
    Message:               kustomize build failed: Failed to read kustomization file under /tmp/kustomization-2223444/releases/testdeploy:
apiVersion for Kustomization should be kustomize.config.k8s.io/v1beta1
    Reason:                 BuildFailed
    Status:                 False
    Type:                   Ready
  Last Attempted Revision:  xyzz/37036033e876f41ffdgdgdgdfg
  Observed Generation:      1
Events:
  Type     Reason       Age   From                  Message
  ----     ------       ----  ----                  -------
  Warning  BuildFailed  48m   kustomize-controller  kustomize build failed: Failed to read kustomization file under /tmp/kustomization-2223444/releases/testdeploy:
apiVersion for Kustomization should be kustomize.config.k8s.io/v1beta1
```

Kustomization YAML contents:

```
apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
kind: Kustomization
metadata:
  name: testdeploy
  namespace: flux-system
spec:
  interval: 5m
  retryInterval: 1m
  timeout: 5m
  path: "./charts/testdeploy/"
  prune: true
  sourceRef:
    kind: GitRepository
    name: testdeploy
    namespace: flux-system
```

**Comment by kingdonb** [comment, marked as answer]
OK, I think I see the issue here. You must be aware that there are two Kustomization types (this is what I call a common tripping hazard on the way into learning about Flux...)

https://fluxcd.io/flux/faq/#are-there-two-kustomization-types

What you have written is a Flux Kustomization, but if you put it into a file `kustomization.yaml` then Kustomize controller will expect it to be a SIG-CLI Kustomization, the other type.

The easy fix is to ensure that your Flux Kustomization is in a file named anything but `kustomization.yaml`

When you create a `kustomization.yaml` file, it blocks Flux from doing the "plain YAML" version – what exactly happens under the hood is a bit complicated, but here is the FAQ about it in case you want the details:

https://fluxcd.io/flux/faq/#what-is-the-behavior-of-kustomize-used-by-flux

tl;dr: when no `kustomization.yaml` is present, Flux will slurp up all the YAML files as you expected it to do.

Hope this helps!

**Reply by cnukwas** [reply]
Thank you so much for this tip., this was it. I did see a note about two Kustomization types but didn't think about naming it differently. I believer it will help to add a note in the [plain yamls](https://fluxcd.io/flux/faq/#can-i-use-repositories-with-plain-yamls) section of the documentation to clarify this.

**Reply by cnukwas** [reply]
> tl;dr: when no `kustomization.yaml` is present, Flux will slurp up all the YAML files as you expected it to do.

In my case I didn't explicitly create another kustomization.yaml other than Flux version of Kustomization as mentioned about with GitRepository reference to let Flux sync from given release/path, is that what you're referring to or we don't need any Kustomization kind when dealing with plain YAMLs?

**Reply by kingdonb** [reply]
> didn't [..] create another `kustomization.yaml` other than Flux version

If you named your file `kustomization.yaml` then it cannot be a Flux kustomization, is what I tried to say. These are automatically interpreted as sig-cli Kustomize overlays, you must name Flux kustomization files something else.

I usually go with `flux-kustomization.yaml` or `nameofapp-kustomization.yaml`
